### ISSUE TITLE
Composer Require Hanging

### ISSUE THREAD

**Issue by RobertGrubb** [composer/composer#8961]
My `composer.json`:

```json
{}
```

Output of `composer diagnose`:

```
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
```

When I run this command:

```
composer require <any_package_here>
```

I get the following output:

```
Nothing, it hangs
```

And I expected this to happen:

To install the package requested, but it's just hanging. I've cleared the composer cache, but that's not helping at all. I've also attempted to test several packages on a remote server as well and am getting the same issue. Really was just curious if anyone else is having this problem.

**Comment by RobertGrubb** [comment]
Nevermind, it magically started working again. >_> lol.

**Comment by pedroborges** [comment]
This started happening today. Running `composer require` with `-vvv` shows it slows down when it hits the first `Downloading https://repo.packagist.org/p/provider`. It eventually finishes its things, but it is taking more than half an hour.

<img width="996" alt="Screen Shot 2020-06-17 at 21 34 34" src="https://user-images.githubusercontent.com/688309/84964671-14b9b100-b0e3-11ea-8d14-bcaaf55f57be.png">

Already tried uninstalling/installing, restarting the computer, restarting the modem, mobile hotspot, same slowness. My internet is fine for anything else.

<img width="824" alt="Screen Shot 2020-06-17 at 21 35 16" src="https://user-images.githubusercontent.com/688309/84964680-1be0bf00-b0e3-11ea-9d71-e2be19af4cd5.png">

Ping from mobile hotspot:

<img width="534" alt="Screen Shot 2020-06-17 at 21 32 08" src="https://user-images.githubusercontent.com/688309/84964691-2438fa00-b0e3-11ea-8554-da73a3ac2d4c.png">

Ping from home fiber:

<img width="533" alt="Screen Shot 2020-06-17 at 21 39 11" src="https://user-images.githubusercontent.com/688309/84964720-39158d80-b0e3-11ea-9d91-d65f1269ce90.png">

**Comment by themainframe** [comment]
Indeed, I attempted a clean install and a simple: `php composer.phar require guzzle/guzzle` today.

Took well over 5 minutes on a fast network connection to pull all the JSONs down. The really annoying thing here (and the thing that I think makes this a Composer issue rather than a packagist issue) is that without `-vvv` there's _absolutely no_ output. See the first line at the bottom of this screenshot - that's the first thing a "non verbose" user would see after 5 minutes twiddling their thumbs... Needs some output if it's going to take that long.

<img width="1359" alt="Screenshot 2020-06-19 at 23 52 36" src="https://user-images.githubusercontent.com/54750/85184603-399b5900-b288-11ea-8add-feefd4ba62b4.png">

**Comment by unzaebk** [comment]
Your are trying to run Composer on a production box, i guess. And you are updating your packages, including the development packages: **require-dev**.

_(A)_ I doubt that you want to do your development on a machine, which such harsh constraints, but anyway, here are some pointers:

reduce the workload for Composer's solver by defining exact versions, instead of relying on additional operators, like tilde **~** or wildcard *****. For example: don't use **1.2.*** or **~1.2**, when there is a **1.2.3.**
skip installing packages listed in require-dev **composer update --no-dev**, you don't need them on a production box
excute composer in verbose mode, to find out how much more memory you would need ;)
But don't be surprised, when it still doesn't work.

_(B)_ To really solve the issue, you might properly package your project on a development machine. Just run **composer install --no-dev** or **composer update --no-dev** on the dev machine and package your project folder, then deploy it to production (copy file, extract).

_(C)_ Run **update** locally, push the **composer.lock** file to Github and then pull it on the production server, then run **install**. By doing that, the already resolved versions from the lock file are used, lowering the work and memory usage.

In short:

> build a release on the dev box, then deploy to production
> this avoids executing Composer on the machine with constraints

**Comment by themainframe** [comment]
Forgive me, but how does any of that solve the problem of an incredibly long delay on first use with no UI feedback as the package manifests are downloaded? That's what I'm complaining about here. FTR this wasn't on a production box, I deliberately simulated a first-time use of composer to replicate this problem.

**Comment by clyer** [comment]
IP 3.248.97.47 not available.

**Comment by Seldaek** [comment]
Please try https://github.com/composer/composer/releases/tag/2.0.0-RC2 - it hopefully is much more responsive for everyone.
